boa_engine/src/object/builtins/jspromise.rs

```diff
@@ -833,6 +891,114 @@ impl JsPromise {
 
         Self::from_object(value.clone())
     }
+
+    /// Creates a `JsFuture` from this `JsPromise`.
+    ///
+    /// The returned `JsFuture` implements [`Future`], which means it can be `await`ed within Rust's
+    /// async contexts (async functions and async blocks).
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use std::error::Error;
+    /// # use boa_engine::{
+    /// #     builtins::promise::PromiseState,
+    /// #     object::builtins::JsPromise,
+    /// #     Context, JsValue, JsError
+    /// # };
+    /// # use futures_lite::future;
+    /// # fn main() -> Result<(), Box<dyn Error>> {
+    /// let context = &mut Context::default();
+    ///
+    /// let (promise, resolvers) = JsPromise::new_pending(context);
+    /// let promise_future = promise.into_js_future(context)?;
+    ///
+    /// let future1 = async move {
+    ///     promise_future.await
+    /// };
+    ///
+    /// let future2 = async move {
+    ///     resolvers.resolve.call(&JsValue::undefined(), &[10.into()], context)?;
+    ///     context.run_jobs();
+    ///     Ok::<(), JsError>(())
+    /// };
+    ///
+    /// let (result1, result2) = future::block_on(future::zip(future1, future2));
+    ///
+    /// assert_eq!(result1, Ok(JsValue::from(10)));
+    /// assert_eq!(result2, Ok(()));
+    ///
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn into_js_future(self, context: &mut Context<'_>) -> JsResult<JsFuture> {
+        // Mostly based from:
+        // https://rustwasm.github.io/wasm-bindgen/api/src/wasm_bindgen_futures/lib.rs.html#109-168
```

Comment:
This link will probably change in the future, we should find a link that doesn't change (to a given version / Git commit)